prototype/mobile_endpoint/form/models.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import absolute_import
+
+import datetime
+import hashlib
+from copy import copy
+
+from jsonobject.api import re_date, JsonObject
+from jsonobject.base import DefaultProperty
+from jsonobject.properties import DateTimeProperty, StringProperty, DictProperty, BooleanProperty, ListProperty
+from mobile_endpoint.form import const
+
+
+def doc_types():
+    """
+    Mapping of doc_type attributes in CouchDB to the class that should be instantiated.
+    """
+    return {
+        'XFormInstance': XFormInstance,
+        # 'XFormArchived': XFormArchived,
+        # 'XFormDeprecated': XFormDeprecated,
+        'XFormDuplicate': XFormDuplicate,
+        'XFormError': XFormError,
+        # 'SubmissionErrorLog': SubmissionErrorLog,
+    }
+
+
+class Metadata(JsonObject):
+    """
+    Metadata of an xform, from a meta block structured like:
+
+        <Meta>
+            <timeStart />
+            <timeEnd />
+            <instanceID />
+            <userID />
+            <deviceID />
+            <deprecatedID /> 
+            <username />
+
+            <!-- CommCare extension -->
+            <appVersion />
+            <location />
+        </Meta>
+
+    See spec: https://bitbucket.org/javarosa/javarosa/wiki/OpenRosaMetaDataSchema
+
+    username is not part of the spec but included for convenience
+    """
+    timeStart = DateTimeProperty()
+    timeEnd = DateTimeProperty()
+    instanceID = StringProperty()
+    userID = StringProperty()
+    deviceID = StringProperty()
+    deprecatedID = StringProperty()
+    username = StringProperty()
+    appVersion = StringProperty()
+    # location = GeoPointProperty()
+
+
+class XFormOperation(JsonObject):
+    """
+    Simple structure to represent something happening to a form.
+
+    Currently used just by the archive workflow.
+    """
+    user = StringProperty()
+    date = DateTimeProperty(default=datetime.datetime.utcnow)
+    operation = StringProperty()  # e.g. "archived", "unarchived"
```

Comment:
Yea, I don't think it gets used.

On 8 July 2015 at 14:30, Cory Zue <notifications@github.com> wrote:

> In prototype/mobile_endpoint/form/models.py
> <https://github.com/dimagi/mobile-endpoint/pull/13#discussion_r34141650>:
>
> > +    deviceID = StringProperty()
> > +    deprecatedID = StringProperty()
> > +    username = StringProperty()
> > +    appVersion = StringProperty()
> > +    # location = GeoPointProperty()
> > +
> > +
> > +class XFormOperation(JsonObject):
> > +    """
> > +    Simple structure to represent something happening to a form.
> > +
> > +    Currently used just by the archive workflow.
> > +    """
> > +    user = StringProperty()
> > +    date = DateTimeProperty(default=datetime.datetime.utcnow)
> > +    operation = StringProperty()  # e.g. "archived", "unarchived"
>
> i totally forgot we had this thing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/mobile-endpoint/pull/13/files#r34141650>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa